### PR TITLE
`CoreConfig` implementation on the managed side

### PR DESCRIPTION
**Pull request by KillStr3aK**
I've 'attached' single comments on some part of the code, let me know if this requires modification.
I'll make the unmanaged implementation aswell, and a documentation for this later on

I'm also working on the standardized plugin configuration behind the scenes but that will come with a different pr as I'm not done with that yet.

### CODE REVIEW

**Comment by zonical** [comment]
The only suggestion I would make is to turn the comments in `ccnfigs/core.json` into a page on the documentation. Some IDE's might not like comments in JSON, like Visual Studio Code by default.

**Comment by KillStr3aK** [comment]
> Some IDE's might not like comments in JSON, like Visual Studio Code by default.
Yes, VSC requires .jsonc to be selected to avoid error highlighting for the comments

> EDIT: Just saw "and a documentation for this later on" - you're one step ahead!
I'll make a doc page for it when I'm done with the configuration standard, also will remove the comments from the `core.json`

File: docs/src/content/docs/configuration/core-configuration.md
Comment (roflmuffin) [inline docs/src/content/docs/configuration/core-configuration.md:1]: I think this file needs to be moved to `docs/reference` folder for it to be auto-picked up by the contents tree.

File: docs/src/content/docs/configuration/core-configuration.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+## FollowCS2ServerGuidelines
+
+Per [CS2 Server Guidelines](https://blog.counter-strike.net/index.php/server_guidelines/), certain plugin
+functionality will trigger all of the game server owner's Game Server Login Tokens
+(GSLTs) to get banned when executed on a Counter-Strike 2 game server.
+
+Enabling this option will block plugins from using functionality that is known to cause this.
+This option only has any effect on CS2. Note that this does NOT guarantee that you cannot
+receive a ban.
```
Comment (roflmuffin) [inline docs/src/content/docs/configuration/core-configuration.md:22]: I think this comment is redundant (probably a copy pasta)
```suggestion
Note that this does NOT guarantee that you cannot
receive a ban.
```

File: managed/CounterStrikeSharp.API/Core/CoreConfig.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        /// <summary>
+        /// <para>
+        /// Per <see href="http://blog.counter-strike.net/index.php/server_guidelines/"/>, certain plugin
+        /// functionality will trigger all of the game server owner's Game Server Login Tokens
+        /// (GSLTs) to get banned when executed on a Counter-Strike 2 game server.
+        /// </para>
+        ///
+        /// <para>
+        /// Enabling this option will block plugins from using functionality that is known to cause this.
+        /// This option only has any effect on CS2. Note that this does NOT guarantee that you cannot
```
Comment (roflmuffin) [inline managed/CounterStrikeSharp.API/Core/CoreConfig.cs:65]:
```suggestion
        /// Note that this does NOT guarantee that you cannot
```

File: docs/src/content/docs/configuration/core-configuration.md (hunk 1 above)
Comment (KillStr3aK) [reply docs/src/content/docs/configuration/core-configuration.md:22]: Committed suggestion, yes it is indeed a reference from SM

File: managed/CounterStrikeSharp.API/Core/CoreConfig.cs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    public static partial class CoreConfig
+    {
+        private static CoreConfigData _coreConfig = new CoreConfigData();
+
+        static CoreConfig()
+        {
+            CommandUtils.AddStandaloneCommand("css_core_reload", "Reloads the core configuration file.", ReloadCoreConfigCommand);
+        }
+
+        [RequiresPermissions("can_reload_core")]
```
Comment (roflmuffin) [inline managed/CounterStrikeSharp.API/Core/CoreConfig.cs:85]:
```suggestion
        [RequiresPermissions("@css/config")]
```

File: docs/src/content/docs/configuration/core-configuration.md
Comment (KillStr3aK) [reply docs/src/content/docs/configuration/core-configuration.md:1]: I've moved the file to the `reference` folder

**Pull request merged**